gcc/rust/parse/rust-parse-impl.h

```diff
@@ -1385,6 +1387,25 @@ Parser<ManagedTokenSource>::parse_vis_item (AST::AttrVec outer_attrs)
 	  lexer.skip_token (1); // TODO: is this right thing to do?
 	  return nullptr;
 	}
+    case ASYNC:
+      t = lexer.peek_token (1);
+
+      switch (t->get_id ())
+	{
+	case UNSAFE:
+	case FN_TOK:
+	  return parse_function (std::move (vis), std::move (outer_attrs));
+
+	default:
+	  add_error (
+	    Error (t->get_locus (),
+		   "unexpected token %qs in some sort of async production",
+		   t->get_token_description ()));
+
+	  lexer.skip_token (1);
+	  return nullptr;
+	}
+
```

Comment:
async is not allowed if the Rust edition is set to 2015, which is the default. I think keeping the parser like that is fine, but we should add that to the checks in the AST validation pass

Reply:
Rustc error out on `async` during parsing. We probably want to do the same to ensure macro compatibility.

Reply:
Could you tell me where or how I can find which version of Rust is set ? I know the magic happens in the lang.opt file, but how do we know what parameters are passed?

Reply:
AFAIK there is a singleton in the rust session manager.